### DISCUSSION TITLE
remarkparse is parsing Link as a Paragraph

### DISCUSSION THREAD

**Discussion by shriharip** [remarkjs/remark#1256] [Q&A] [answered]
Dear all,

I am trying to build a new plugin and startd with development.

````
import remarkParse from 'remark-parse'
import rehypeStringify from 'rehype-stringify'
import {unified} from 'unified'
import  myPlugin from './plugin.js';

const file = await unified()
.use(remarkParse)
.use(myPlugin, {})
.use(rehypeStringify)
.process(`[mylink] ( https://example.com "link title") `)

console.log(file);
```

inside the myPlugin file, whcih is just a return function for now

````
return async function (tree, _) => {

vvisit(tree, (node) => {
      console.log(node)
    });
}
```

The link is coming as a paragraph. 
````
{
  type: 'paragraph',
  children: [
    {
      type: 'text',
      value: '[mylink] ( https://example.com "link title")',
      position: [Object]
    }
  ],
  position: {
    start: { line: 1, column: 1, offset: 0 },
    end: { line: 1, column: 32, offset: 31 }
  }
}
````

What am I missing here? 

if I use the `fromMarkdown`of the `mdast-util`it works properly.

**Comment by ChristianMurphy** [comment]
Welcome @shriharip! 👋 
Sorry you ran into some confusion.

The issue is that
```md
[mylink] ( https://example.com "link title")
```
is not a link.
example on GitHub:

---

[mylink] ( https://example.com "link title")

---

Commonmark does not allow spaces between `]` and `(`
this is a valid link:

```md
[mylink](https://example.com "link title")
```

**Comment by shriharip** [comment, marked as answer]
Sorry, found the issue. I had a space between the brackets.
